### PR TITLE
Merge project config & overrides correctly.

### PR DESCRIPTION
**Pull request by BenBirt**
related to https://github.com/dataform-co/dataform-web-tracking/issues/287

This isn't actually the cause of that bug, because the broken merge that bug is referring to happens at a higher level, but I'm fixing this up for completeness.

### CODE REVIEW

File: api/commands/compile.ts [hunk 1]
```diff
@@ -129,7 +123,7 @@ export class CompileChildProcess {
   }
 }
 
-export const checkDataformJsonValidity = (dataformJsonParsed: { [prop: string]: string }) => {
+export const checkDataformJsonValidity = (dataformJsonParsed: { [prop: string]: any }) => {
```
Comment (GJMcGowan) [inline api/commands/compile.ts:126]: seems a shame to lose the typing

File: core/gen_index.ts [hunk 2]
```diff
@@ -62,7 +62,7 @@ projectConfig.schemaSuffix = "${
   }" || projectConfig.schemaSuffix;
 
 // Merge in general project config overrides.
-projectConfig = { ...projectConfig, ...${projectOverridesJsonString} };
+projectConfig = { ...projectConfig, ...${projectOverridesJsonString}, vars: { ...projectConfig.vars, ...${projectOverridesJsonString}.vars } };
```
Comment (GJMcGowan) [inline core/gen_index.ts:65]: Why not deepmerge here as well?

**Review by GJMcGowan — APPROVED** [review]
Same comment about the library as on the other PR.

File: api/commands/compile.ts (hunk 1 above)
Comment (BenBirt) [reply api/commands/compile.ts:126]: the typing isn't actually correct :P

File: core/gen_index.ts (hunk 2 above)
Comment (BenBirt) [reply core/gen_index.ts:65]: This is effectively the index.js that we generate for each DF project. Plumbing through dependencies on other NPM packages at this stage is a bit of a pain, while this isn't ideal, I think it's the best option we have for now.

**Pull request merged**